src/app/customer-item/customer-item-detail/customer-item-detail.component.ts

```diff
@@ -53,22 +62,52 @@ export class CustomerItemDetailComponent implements OnInit {
 				this.getItem();
 			})
 			.catch((getCustomerItemError) => {
-				console.log("hello there", this.getCustomerItem);
 				console.log(
 					"customerItemDetailComponent: could not get customerItem",
 					getCustomerItemError
 				);
 			});
 	}
 
+	private toPrettyDate(date: Date): string {
+		return moment(date).format("DD.MM.YYYY");
+	}
+
 	private getItem() {
 		this._itemService
 			.getById(this.customerItem.item as string)
 			.then((item: Item) => {
 				this.item = item;
+				this.deadlineInput = this.toPrettyDate(
+					this.customerItem.deadline
+				);
 			})
 			.catch((getItemError) => {
 				console.log("customerItemDetailComponet: could not get item");
 			});
 	}
+
+	public hasChangedDate(): boolean {
+		return (
+			this.toPrettyDate(this.customerItem.deadline) !== this.deadlineInput
+		);
+	}
+
+	public async handleUpdateDeadline() {
+		try {
+			this.customerItem = await this._customerItemService.update(
+				this.customerItem.id,
+				{
+					deadline: moment(this.deadlineInput, "DD.MM.YYYY"),
```

Comment:
This is logged when you try to save an invalid deadline; I guess it's fine, though I would prefer not trying to update if moment returns null.
```
silly -> {"deadline":null}
error failed to save document: ValidationError: deadline: Path `deadline` is required.
verbose ! (701): failed to save the document
verbose ! (701) {"deadline":null}
verbose ! (701) {"errors":{"deadline":{"name":"ValidatorError","message":"Path `deadline` is required.","properties":{"message":"Path `deadline` is required.","type":"required","path":"deadline","value":null},"kind":"required","path":"deadline","value":null}},"_message":"customeritems validation failed","name":"ValidationError","message":"customeritems validation failed: deadline: Path `deadline` is required."}
```

Reply:
oh well, that is kinda how things are handled in bl-admin. In bl-next admin, we can improve